src/tree.rs

```diff
@@ -0,0 +1,43 @@
+use crate::event::Event;
+use crate::query::Queryable;
+use crate::Node;
+use accesskit::TreeUpdate;
+use parking_lot::Mutex;
+
+pub struct Tree {
+    tree: accesskit_consumer::Tree,
+    queued_events: Mutex<Vec<Event>>,
+}
+
+pub(crate) type EventQueue = Mutex<Vec<Event>>;
+
+impl Tree {
+    pub fn new(update: TreeUpdate) -> Self {
+        Self {
+            tree: accesskit_consumer::Tree::new(update, true),
+            queued_events: Mutex::new(Vec::new()),
+        }
+    }
+
+    pub fn update(&mut self, update: accesskit::TreeUpdate) {
+        self.tree.update(update);
+    }
+
+    pub fn root(&self) -> Node<'_> {
+        self.node()
+    }
+
+    pub fn take_events(&self) -> Vec<Event> {
+        self.queued_events.lock().drain(..).collect()
```

Comment:
Why is this `&self` and not `&mut self` (i.e. why is there a Mutex)?

Reply:
I wanted the query functions to take &self, so I don't think a Node can hold a mutable reference to the event queue.
If the nodes would hold mutable references I think you would get problems when querying multiple nodes. 

I think I could replace this with a RefCell though.